pkg/controllers/jobset_controller.go

```diff
@@ -541,6 +508,40 @@ func genSubdomain(js *jobset.JobSet, rjob *jobset.ReplicatedJob) string {
 	return fmt.Sprintf("%s-%s", js.Name, rjob.Name)
 }
 
+func genLabels(js *jobset.JobSet, rjob *jobset.ReplicatedJob, job *batchv1.Job, jobIdx int) map[string]string {
+	labels := make(map[string]string)
+
+	// Add jobset name as label to pod template spec.
```

Comment:
all comments in this and the genAnnotations func are not really adding value, the code is self explanatory. I would also just put all of this inside labelAndAnnotate func since they are always used together.